mlflow/transformers/hub_utis.py

```diff
@@ -0,0 +1,51 @@
+import functools
+import logging
+import os
+
+from mlflow.exceptions import MlflowException
+from mlflow.protos.databricks_pb2 import RESOURCE_DOES_NOT_EXIST
+
+_logger = logging.getLogger(__name__)
+
+
+@functools.lru_cache(maxsize=1)
+def get_latest_commit_for_repo(repo: str) -> str:
```

Comment:
Is `maxsize` correct? We don't call this function multiple times with different repos?

Reply:
Most of the use case should be single repo, unless the user mixes components from multiple repos, or logging different pipelines, which I think is rare. Indeed, the caching here is mainly for repeated fetch in one save call - we don't want to persist it beyond that scope, cuz the cached commit hash might become stale. However, I didn't know if there is any easy way to enforce the refresh, so instead just set `1` to at least trigger refresh when called with different repo... This is obviously not a sufficient solution so let me know if you have better idea:)

Reply:
Makes sense. Can we insert a comment explaining why maxsize is set to 1?

Reply:
Sure!